modules/createServerRenderContext.js

```diff
@@ -1,34 +1,36 @@
-const k = () => {}
-
 const createServerRenderContext = () => {
   let flushed = false
+  let flushedIndex = 0
   let redirect = null
   let matchContexts = [
-    /* { hasMissComponent: bool, matchesByIdentity: [] } */
+    /* { hasMissComponent: bool, hasMatches: bool } */
   ]
 
-  const setRedirect = flushed ? k : (location) => {
+  const setRedirect = (location) => {
     if (!redirect)
       redirect = location
   }
 
-  const registerMatchContext = flushed ? k : (matchesByIdentity) => {
-    return matchContexts.push({
-      hasMissComponent: false,
-      matchesByIdentity
-    }) - 1
+  // on the second pass, index is determined by render order
+  const registerMatchProvider = () => {
+    return flushed ? flushedIndex++ : (
```

Comment:
Am I correct here in believing that the order in which `<MatchProvider>`s will be rendered is always the same for the first and second pass?